shell/platform/darwin/ios/framework/Source/FlutterPlatformViewsTest.mm

```diff
@@ -1816,17 +1814,42 @@ - (void)testClipRRect {
   [mockFlutterView setNeedsLayout];
   [mockFlutterView layoutIfNeeded];
 
+  /*
+  ClippingMask          outterClipping
+    2 3 4 5 6 7           2 3 4 5 6 7
+  2 / - - - - \         2 + - - - - +
+  3 |         |         3 |         |
+  4 |         |         4 |         |
+  5 |         |         5 |         |
+  6 |         |         6 |         |
+  7 \ - - - - /         7 + - - - - +
+
+  innerClipping1        innerClipping2
+    2 3 4 5 6 7           2 3 4 5 6 7
+  2   + - - +           2
+  3   |     |           3 + - - - - +
+  4   |     |           4 |         |
+  5   |     |           5 |         |
+  6   |     |           6 + - - - - +
+  7   + - - +           7
+  */
+  CGRect innerClipping1 = CGRectMake(3, 2, 4, 6);
+  CGRect innerClipping2 = CGRectMake(2, 3, 6, 4);
+  CGRect outterClipping = CGRectMake(2, 2, 6, 6);
   for (int i = 0; i < 10; i++) {
     for (int j = 0; j < 10; j++) {
       CGPoint point = CGPointMake(i, j);
       int alpha = [self alphaOfPoint:CGPointMake(i, j) onView:mockFlutterView];
-      // Edges of the clipping might have a semi transparent pixel, we only check the pixels that
-      // are fully inside the clipped area.
```

Comment:
Again the existing behavior is incorrect. 

In the picture below, the **black rounded rect** is the mask; the **red rect** is `innerClipping1`, and **green rect** is `innerClipping2`. 

Pixels within the red or green rect should be fully opaque; pixels outside of `outterClipping` should be fully transparent; the 4 corner pixels should be partially transparent. 

![IMG_7331](https://github.com/flutter/engine/assets/41930132/f0bee6e4-b559-4402-95eb-daaf12ed1cdd)